src/vs/vscode.d.ts

```diff
@@ -1605,9 +1605,15 @@ declare module 'vscode' {
 		scheme?: string;
 
 		/**
-		 * A glob pattern, like `*.{ts,js}`.
+		 * Either a glob pattern, like `**∕*.{ts,js}` or a [workspace pattern](#WorkspacePattern) that can be relative
+		 * and matches on the path of the provided workspace.
 		 */
-		pattern?: string;
+		pattern?: string | WorkspacePattern;
+	}
+
+	export interface WorkspacePattern {
```

Comment:
jsdoc missing

Reply:
I wonder if we should also make it a relative pattern as we have in the lower level. A `class` like this

```ts
class RelativePattern {
  base: Uri;
  pattern: string;

  constructor(pattern:string, base: WorkspaceFolder | Uri)
}
```

The ctor adds some util-sugar to it and making just a `Uri` allows for more generic use in extensions, e.g watch for file event in a certain folder (unrelated to a workspace folder)

Reply:
Yeah I like that 👍

Reply:
In addition I would add the type to createFileSystemWatcher. Eg. globPattern: string | RelativePattern